File: packages/logger-android/src/main/kotlin/com/codex/logger/LoggerConfig.kt
package com.codex.logger

data class LoggerConfig(
    val storagePath: String,
    val minLogLevel: LogLevel = LogLevel.INFO,
    val maxDiskBytes: Long = 20L * 1024L * 1024L,
    val maxSegmentBytes: Long = 1L * 1024L * 1024L,
    val flushIntervalMs: Long = 5_000L,
    val bufferSize: Int = 20,
    val enableConsoleOutput: Boolean = false,
    val enableThreadInfo: Boolean = true,
    val redactedKeys: Set<String> = setOf("password", "token", "authorization"),
    val redactionPatterns: List<Regex> = emptyList(),
)

File: packages/logger-ios/Sources/LoggerConfig.swift
public struct LoggerConfig: Sendable {
    public let storagePath: String
    public let minLogLevel: LogLevel
    public let maxDiskBytes: Int64
    public let maxSegmentBytes: Int64
    public let flushIntervalMs: Int
    public let bufferSize: Int
    public let enableConsoleOutput: Bool
    public let enableThreadInfo: Bool
    public let redactedKeys: Set<String>
    public let redactionPatterns: [String]

    public init(
        storagePath: String,
        minLogLevel: LogLevel = .info,
        maxDiskBytes: Int64 = 20 * 1024 * 1024,
        maxSegmentBytes: Int64 = 1 * 1024 * 1024,
        flushIntervalMs: Int = 5_000,
        bufferSize: Int = 20,
        enableConsoleOutput: Bool = false,
        enableThreadInfo: Bool = true,
        redactedKeys: Set<String> = ["password", "token", "authorization"],
        redactionPatterns: [String] = []
    ) {
        self.storagePath = storagePath
        self.minLogLevel = minLogLevel
        self.maxDiskBytes = maxDiskBytes
        self.maxSegmentBytes = maxSegmentBytes
        self.flushIntervalMs = flushIntervalMs
        self.bufferSize = bufferSize
        self.enableConsoleOutput = enableConsoleOutput
        self.enableThreadInfo = enableThreadInfo
        self.redactedKeys = redactedKeys
        self.redactionPatterns = redactionPatterns
    }
}
